### PR TITLE
disable padding[0] check for conv3d, add test config

### LINKED ISSUE #25936: [Bounty $1500] Conv3d T dim padding
## 📝 Background

Currently, `ttnn.experimental.conv3d` in tt-metal does not support padding along the T (temporal) dimension, which is a standard parameter in 3D convolution operations. This limitation is enforced by an assertion that requires `padding[0] == 0`, effectively blocking temporal padding. Supporting padding along the T dimension is crucial for applications like video analysis, volumetric medical imaging, and 3D object detection—where proper temporal context is necessary for correctness and generality.

## 🎯 What Success Looks Like

- Remove the restriction that forces `padding[0] == 0` in the device code.
- Update kernel implementation to correctly handle non-zero T-dimension (temporal) padding.
- Output dimension calculations account for T-dimension padding.
- API accepts the same padding parameter format as PyTorch's `torch.nn.Conv3d`, including T-dimension (D in PyTorch) padding.
- Support both "zeros" and "replicate" padding modes for the T dimension.
- Comprehensive tests added to `tests/ttnn/nightly/unit_tests/operations/conv/test_conv3d.py` validating numerical correctness against PyTorch Conv3d for several non-trivial temporal padding cases.

## 💡 Problem to Solve

The current tt-metal Conv3d device implementation blocks T-dimension padding by assertion, even though the underlying config and kernel structure could support it. PyTorch’s Conv3d supports arbitrary padding, including the depth/temporal dimension. This bounty requires updating the device operation and kernel logic to permit and correctly process temporal padding—making the implementation correct and much more broadly applicable.

## 🧭 Guidance & Starting Points

- See the assertion in `ttnn/cpp/ttnn/operations/experimental/conv3d/device/conv3d_device_operation.cpp`:
  ```cpp
  // assert padding on T is zero
  TT_FATAL(
      config.padding[0] == 0,
      "Padding must be (0,x,x). got ({}, {}, {})",
      config.padding[0],
      config.padding[1],
      config.padding[2]);
  ```
- Kernel logic for vol2col in `ttnn/cpp/ttnn/operations/experimental/conv3d/device/reader_vol2col.cpp` already has a framework for padding handling, but is restricted - ensure T-dimension logic is complete.
- Output shape calculation must include T-dimension padding.
- Test coverage in `tests/ttnn/nightly/unit_tests/operations/conv/test_conv3d.py`—add cases such as:
  - `padding=(1,1,1)`, `padding=(2,0,1)`, `padding=(3,2,1)`
  - Both "zeros" and "replicate" T-dimension padding
  - Compare output to PyTorch Conv3d

## 🔎 Possible Approaches

- Remove the assertion enforcing zero T padding.
- Read and update the kernel logic for proper handling of temporal padding. Follow the pattern for H/W padding.
- Refactor or extend output dimension calculation to match PyTorch’s convention.
- Use PyTorch Conv3d as reference for expected behavior and output.
- Write/extend unit tests to confirm correctness for typical and edge-case padding values.

---
## 📚 Resources

- [PyTorch Conv3d Documentation](https://pytorch.org/docs/stable/generated/torch.nn.Conv3d.html)
- Existing Conv2d implementation for reference patterns in padding logic
- Relevant tt-metal files:
  - [`conv3d_device_operation.cpp`](https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/experimental/conv3d/device/conv3d_device_operation.cpp)
  - [`reader_vol2col.cpp`](https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/experimental/conv3d/device/reader_vol2col.cpp)
  - [`test_conv3d.py`](https://github.com/tenstorrent/tt-metal/blob/main/tests/ttnn/nightly/unit_tests/operations/conv/test_conv3d.py)

### PR DESCRIPTION
**Pull request by RafLit**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/25936)

### Problem description
Currently, ttnn.experimental.conv3d in tt-metal does not support padding along the T (temporal) dimension, which is a standard parameter in 3D convolution operations. This limitation is enforced by an assertion that requires padding[0] == 0, effectively blocking temporal padding. Supporting padding along the T dimension is crucial for applications like video analysis, volumetric medical imaging, and 3D object detection—where proper temporal context is necessary for correctness and generality.

### What's changed
- removed the assertion in ttnn/cpp/ttnn/operations/experimental/conv3d/device/conv3d_device_operation.cpp verifying that temporal dimension padding is equal to zero
- verified that the kernel logic in ttnn/cpp/ttnn/operations/experimental/conv3d/device/reader_vol2col.cpp file is complete
- added test configuration for various cases of temporal dimension padding in tests/ttnn/nightly/unit_tests/operations/conv/test_conv3d.py

### Checklist
- [X] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI [passes](https://github.com/tenstorrent/tt-metal/actions/runs/22132309027)
- [X] [Nightly tt-metal L2 tests](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml) CI [passes](https://github.com/tenstorrent/tt-metal/actions/runs/21991142685)

### CODE REVIEW

**Comment by astancovTT** [comment]
Hello, I will be doing the review for this PR.
I've run 2 pipelines for this branch:
All post commit: https://github.com/tenstorrent/tt-metal/actions/runs/20615188686
Nightly L2 (conv): https://github.com/tenstorrent/tt-metal/actions/runs/20615194555

**Comment by astancovTT** [comment]
please rebase to latest main

**Comment by astancovTT** [comment]
lastest l2 run: https://github.com/tenstorrent/tt-metal/actions/runs/21067370458

**Comment by jberkowitzTT** [comment]
@RafLit - I wanted to bring the latest comment to your attention. Let us know if you need any support in completing this bounty

**Review by astancovTT — APPROVED** [review]
